src/app/routers/dataset_router.py

```diff
@@ -90,8 +90,8 @@ async def get_unit_supplementary_data(
     },
 )
 async def get_dataset_metadata_collection(
-    survey_id: str = None,
```

Comment:
you know how this is done in CIR with the request objects, should we look at implementing that here in future? 

Also some inconsistency where you've not done | None = None for all, whats the rule its failing for reference?

Reply:
Without `| None` it's implied optional which PEP484 prohibits https://docs.astral.sh/ruff/rules/implicit-optional/